mat_dp_pipeline/sdf/standard_data_format.py

```diff
@@ -117,49 +112,78 @@ def is_leaf(self) -> bool:
         return self.targets is not None
 
     def validate(self) -> None:
-        def has(item) -> bool:
-            return item is not None
+        def validate_yearly_keys(base: pd.DataFrame, yearly: dict[Year, pd.DataFrame]):
+            base_keys = set(base.index.unique())
+
+            for year, df in yearly.items():
+                validate(
+                    set(df.index.unique()) <= base_keys,
+                    f"{self.name}: Yearly file ({year}) introduces new items!",
+                )
 
         validate(
-            has(self.targets) != (len(self.children) > 0),
-            "SDF must either have children in the hierarchy or defined targets (leaf level)",
+            (self.targets is not None) != (len(self.children) > 0),
+            f"{self.name}: SDF must either have children in the hierarchy or defined targets (leaf level)",
         )
         validate(
-            has(self.intensities) or not has(self.intensities_yearly),
-            "No base intensities, while yearly files provided!",
+            (self.intensities is not None) or (self.intensities_yearly is None),
+            f"{self.name}: No base intensities, while yearly files provided!",
         )
         validate(
-            has(self.indicators) or not has(self.indicators_yearly),
-            "No base indicators, while yearly files provided!",
+            (self.indicators is not None) or (self.indicators_yearly is None),
```

Comment:
I think this is a bug based on previous values - should be:
(self.indicators is not None) or (self.indicators_yearly is not None),

Reply:
ah nvm missed a not